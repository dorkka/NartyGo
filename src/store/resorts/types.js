export const SET_RESORTS = 'resorts/SET_RESORTS';
export const IS_LOADING = 'resorts/IS_LOADING';
export const SET_ERROR = 'resorts/SET_ERROR';
export const SET_RESORT = 'resorts/SET_RESORT';
export const SET_SELECTED_OPTIONS = 'SET_SELECTED_OPTIONS';
