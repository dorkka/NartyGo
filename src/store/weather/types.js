export const SET_WEATHER = 'weather/SET_WEATHER';
export const IS_LOADING = 'weather/IS_LOADING';
export const SET_ERROR = 'weather/SET_ERROR';
