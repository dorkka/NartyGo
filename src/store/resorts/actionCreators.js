import * as types from './types';
import resourceFetcher from '../../services/resourceFetcher';

export const setResorts = (data, headers, perPage) => {
  const pageCount = Math.ceil(headers.get('x-total-count') / perPage);
  const list = [];
  const byId = {};
  data.forEach(element => {
    list.push(element.id);
    byId[element.id] = element;
  });
  return {
    type: types.SET_RESORTS,
    payload: {
      pageCount,
      list,
      byId,
    },
  };
};

export const setError = (error) => ({
  type: types.SET_ERROR,
  error,
});

export const setIsLoading = () => ({
  type: types.IS_LOADING,
});

export const setSpecificResort = data => {
  const byId = { [data.id]: data };
  return {
    type: types.SET_RESORT,
    payload: {
      byId,
    },
  };
};

export const getResorts = (_page, _limit) => (dispatch) => {
  dispatch(setIsLoading());
  return resourceFetcher('http://localhost:3001/resorts')({ _page, _limit })
    .then(({ data, headers }) => {
      dispatch(setResorts(data, headers, _limit));
    })
    .catch(error => dispatch(setError(error)));
};

export const getSpecificResort = (id) => (dispatch, getState) => {
  if ((getState().resorts.byId[id])) {
    return null;
  }
  dispatch(setIsLoading());
  return resourceFetcher(`http://localhost:3001/resorts/${id}`)()
    .then(({ data }) => {
      dispatch(setSpecificResort(data));
    })
    .catch(error => dispatch(setError(error)));
};

export const setSelectOptions = (data) => {
  const resortsOptions = []
  const cityOptions = []
  let label = ''
  data.forEach((element) => {
    label = `${element.name} - ${element.city}`;
    resortsOptions.push({ value: element.name, label, id: element.id })
    cityOptions.push({ value: element.city, label, id: element.id })
  })
  return {
    type: types.SET_SELECTED_OPTIONS,
    payload: {
      resortsOptions,
      cityOptions,
    }
  }
}

export const getAllResorts = () => (dispatch) => {
  resourceFetcher('http://localhost:3001/resorts')().then(({ data }) => {
    dispatch(setSelectOptions(data))
  });
}
