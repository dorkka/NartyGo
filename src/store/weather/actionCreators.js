import * as types from './types';
import resourceFetcher from '../../services/resourceFetcher';

export const setWeather = (data) => {
  const weatherTime = new Date().getTime;
  return {
    type: types.SET_WEATHER,
    payload: {
      data,
      weatherTime,
    },
  };
};
export const setError = (error) => ({
  type: types.SET_ERROR,
  error,
});

export const setIsLoading = () => ({
  type: types.IS_LOADING,
});

export const getWeather = (cityId) => (dispatch, getState) => {
  if ((getState().weather.weatherTime - (new Date().getTime)) < 7200000) {
    return;
  }
  dispatch(setIsLoading());
  resourceFetcher(`http://api.openweathermap.org/data/2.5/group?id=${cityId}&lang=pl&units=metric&APPID=07751527bfe01ed870793d6731eefdf1`)()
    .then(({ data }) => {
      dispatch(setWeather(data));
    })
    .catch(error => dispatch(setError(error)));
};
