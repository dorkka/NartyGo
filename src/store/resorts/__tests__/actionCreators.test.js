import * as actionCreators from '../actionCreators';
import * as types from '../types';
