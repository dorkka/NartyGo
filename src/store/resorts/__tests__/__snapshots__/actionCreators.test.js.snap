// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getResorts should call setError when resourceFetcher was called and no data was fetched 1`] = `
Array [
  Object {
    "error": [Error: Something went wrong ...],
    "type": "resorts/SET_ERROR",
  },
]
`;

exports[`getResorts should call setIsLoading when dispatch was called first time 1`] = `
Array [
  Object {
    "type": "resorts/IS_LOADING",
  },
]
`;

exports[`getResorts should call setResorts when all data is fetched 1`] = `
Array [
  Object {
    "payload": Object {
      "byId": Object {
        "1": Object {
          "city": "Ustroń",
          "cityId": 7531846,
          "coordinates": Object {
            "lat": 49.6833162,
            "lng": 18.8196089,
          },
          "id": 1,
          "name": "Czantoria",
          "piste": Array [
            Object {
              "id": 1,
              "name": "red",
              "number": 1,
            },
            Object {
              "id": 2,
              "name": "blue",
              "number": 2,
            },
          ],
        },
        "2": Object {
          "city": "Karlików",
          "cityId": 7531370,
          "coordinates": Object {
            "lat": 49.43529,
            "lng": 22.067767,
          },
          "id": 2,
          "name": "Karlików",
          "piste": Array [
            Object {
              "id": 1,
              "name": "Mała Palenica",
              "number": 1,
            },
          ],
        },
        "3": Object {
          "city": "Szczyrk",
          "cityId": 7532564,
          "coordinates": Object {
            "lat": 49.6685428,
            "lng": 18.9609749,
          },
          "id": 3,
          "name": "Biały Krzyż\\"",
          "piste": Array [
            Object {
              "id": 1,
              "name": "red",
              "number": 1,
            },
            Object {
              "id": 2,
              "name": "green",
              "number": 2,
            },
          ],
        },
      },
      "list": Array [
        1,
        2,
        3,
      ],
      "pageCount": 1,
    },
    "type": "resorts/SET_RESORTS",
  },
]
`;

exports[`getSpecificResort should call setError when resourceFetcher was called and no data was fetched 1`] = `
Array [
  Object {
    "error": [Error: Something went wrong ...],
    "type": "resorts/SET_ERROR",
  },
]
`;

exports[`getSpecificResort should call setIsLoading when dispatch was called first time 1`] = `
Array [
  Object {
    "type": "resorts/IS_LOADING",
  },
]
`;

exports[`getSpecificResort should call setSpecificResort if resort is not in state and data is fetched 1`] = `
Array [
  Object {
    "payload": Object {
      "byId": Object {
        "2": Object {
          "city": "Karlików",
          "cityId": 7531370,
          "coordinates": Object {
            "lat": 49.43529,
            "lng": 22.067767,
          },
          "id": 2,
          "name": "Karlików",
          "piste": Array [
            Object {
              "id": 1,
              "name": "Mała Palenica",
              "number": 1,
            },
          ],
        },
      },
    },
    "type": "resorts/SET_RESORT",
  },
]
`;
